enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,710 @@
+    `workload.openshift.io/target=management` label in a namespace
+    with the `workload.openshift.io/allowed=management` label.
+11. The admission hook modifies the pod, replacing the CPU requests
+    with `cpu.workload.openshift.io/management` requests and adding
+    the `cpu.workload.openshift.io/{container-name}` annotations for
+    CRI-O.
+12. The scheduler sees the new pod and finds available
+    `cpu.workload.openshift.io/management` resources on the node. The
+    scheduler places the pod on the node.
+13. Repeat steps 10-12 until all pods are running.
+14. Single-node deployment comes up with management components
+    constrained to subset of available CPUs
+
+#### Pod mutation
+
+The kubelet and API admission hook will change pods labeled with
+`workload.openshift.io/management` so the CPU requests are replaced
+with management CPU requests and an annotation is added with the same
+value.
+
+```yaml
+requests:
+  cpu:
+    400m
+```
+
+becomes
+
+```yaml
+requests:
+  cpu.workload.openshift.io/management: 400m
+```
+
+and
+
+```yaml
+annotations:
+  io.openshift.workload.management.cpu/container-name: 400m
+```
+
+As a special case, kubelet and the API admission hook should strip any
+annotations from pods trying to set the "cpuset" resource. See the
+next section for more details.
+
+#### CRI-O Changes
+
+*NOTE: This section needs to be updated to deal with the generalized
+CPU pools.*
+
+CRI-O will be updated to support new configuration settings for workload types.
+
+```ini
+[crio.runtime.workloads.{workload-type}]
+  label = "workload.openshift.io/target={workload-type}"
+  resource_annotation_prefix = "io.openshift.workload.{workload-type}"
+  resources = {
+    "cpu": "",
+    "cpuset": "0-1",
+  }
+```
+
+The `label` field is used to match pods that should be treated as
+having the workload type. The label key on the pod is compared for an
+exact match against the value specified in the configuration file.
+
+The `resource_annotation_prefix` is the start of the annotation key
+used to pass settings down to CRI-O.
+
+The `resources` map associates annotation suffixes with default
+values. CRI-O will define a well-known set of resources and other
+values will not be allowed. In OpenShift, we do not want to allow
+arbitrary pods to set their own `cpuset` but other CRI-O users may
+want that ability.
+
+To pass a setting into CRI-O, the pod should have an annotation made
+by combining the `resource_annotation_prefix`, the key from
+`resources`, and the container name, like this:
+
+```
+io.openshift.workload.management.cpu/container_name = {value}
+```
+
+In the management workload case, we will configure it with values like
+
+```ini
+[crio.runtime.workloads.management]
+  label = "workload.openshift.io/target=mangement"
+  resource_annotation_prefix = "io.openshift.workload.management"
+  resources = {
+    "cpu": "",
+    "cpuset": "0-1",
+  }
+```
+
+CRI-O will be configured to support a new annotation on pods,
```

Comment:
@haircommander @mrunalp @browsell This example doesn't seem to match what we have above. Is it still right? Do we still need `allowed_annotations`? Should it have a different value in the example?

Reply:
It should match the latest iteration of what we discussed today for the management stanza.